### PR TITLE
Features/#1103 console settings

### PR DESCRIPTION
**Pull request by barthanssens**
This PR addresses GitHub issue: eclipse/rdf4j#1103 .

Briefly describe the changes proposed in this PR:

* Created interface + separate classes per console setting/parameter (similar to console commands)
* Deprecated the ConsoleParameters interface
* Moved some initialization code in Console.java to separate class for readability
* Some basic testing

Not completely finished yet, but already gives a good impression in what direction I'm thinking.

The code can be cleaned up a bit more, but that would probably require some changes in the signatures or interfaces (e.g. passing a collection of Settings instead of a ConsoleParameters to a few Commands), and that's probably "not done" in a minor release (even though there's probably not much external code out there extending the current console classes / interfaces anyway)

### CODE REVIEW

File: console/src/main/java/org/eclipse/rdf4j/console/Setting.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	/**
+	 * Clear setting
+	 */
+	public void clear();
+	
+	
+	/**
+	 * Load setting from config file
+	 */
+	public void load();
```
[the remark this answers is not in the document]
Comment (barthanssens) [reply console/src/main/java/org/eclipse/rdf4j/console/Setting.java]: Not implemented currently, currently the console seems to create a config directory (.RDF4J, in capitals, on Linux), but doesn't do much with it, other than storing a pretty basic logback.xml. It doesn't even remember settings (currently there is not much to set anyway..)

Probably the easiest to do is to create a user preference file in that directory.

File: console/src/main/java/org/eclipse/rdf4j/console/command/SetParameters.java [hunk 2]
```diff
@@ -116,16 +95,23 @@ private void showAllParameters() {
...
-			showPrefix();
-		} else if (QUERYPREFIX_COMMAND.equalsIgnoreCase(key)) {
-			showQueryPrefix();
-		} else {
-			consoleIO.writeError("unknown parameter: " + key);
+		String str = key.toLowerCase();
+		
+		switch(str) {
+			case LogLevel.NAME:
+				showLogLevel();
```
[the remark this answers is not in the document]
Comment (barthanssens) [reply console/src/main/java/org/eclipse/rdf4j/console/command/SetParameters.java]: Could indeed be improved, I tried to mimize the changes in the current code, but perhaps I should be more brutal.

File: console/src/main/java/org/eclipse/rdf4j/console/setting/DefaultConsoleParameters.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		this.settings = new HashMap<>();
+		settings.put(ConsoleWidth.NAME, new ConsoleWidth(80));
+		settings.put(QueryPrefix.NAME, new QueryPrefix(true));
+		settings.put(ShowPrefix.NAME, new ShowPrefix(true));
+	}
+
+	
+	@Override
+	public int getWidth() {
+		return (Integer) settings.get(ConsoleWidth.NAME).get();
```
[the remark this answers is not in the document]
Comment (barthanssens) [reply console/src/main/java/org/eclipse/rdf4j/console/setting/DefaultConsoleParameters.java]: The goal is to remove the switch and the DefaultConsoleParameters completely, which reduces the complexity and allows for more useful settings

But that would mean changing the argument of some of the commands constructors from accepting a ConsoleParameters to, say, a Map of Settings.

The console commands are not actually part of the RDF4J core API, so I guess it won't break much existing 3rd party code (if it exists at all), just not sure if it is acceptable or not in a minor release.

[the remark this answers is not in the document]
Comment (hmottestad) [reply console/src/main/java/org/eclipse/rdf4j/console/setting/DefaultConsoleParameters.java]: I reckon your code makes things better. So merging is well worth it. Wouldn’t be great to break potentially existing usage no.

File: console/src/main/java/org/eclipse/rdf4j/console/command/SetParameters.java [hunk 4]
```diff
@@ -116,16 +95,23 @@ private void showAllParameters() {
...
-			showWidth();
-		} else if (SHOWPREFIX_COMMAND.equalsIgnoreCase(key)) {
-			showPrefix();
-		} else if (QUERYPREFIX_COMMAND.equalsIgnoreCase(key)) {
-			showQueryPrefix();
-		} else {
-			consoleIO.writeError("unknown parameter: " + key);
+		String str = key.toLowerCase();
+		
+		switch(str) {
```
[the remark this answers is not in the document]
Comment (barthanssens) [reply console/src/main/java/org/eclipse/rdf4j/console/command/SetParameters.java]: Modern java for the win

File: console/src/main/java/org/eclipse/rdf4j/console/Setting.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (barthanssens) [reply console/src/main/java/org/eclipse/rdf4j/console/Setting.java]: I've noticed that the o.e.r.common.app.AppConfiguration actually loads an application.properties file, so I'll use that one

**Comment by barthanssens** [comment]
Thanks, I have still some work to do to get the loading/storing of  the properties to work.
Once that is done, I'll merge it into the dev branch.

File: console/src/main/java/org/eclipse/rdf4j/console/setting/DefaultConsoleParameters.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (barthanssens) [reply console/src/main/java/org/eclipse/rdf4j/console/setting/DefaultConsoleParameters.java]: Meanwhile I've cleaned up and reworked the code a bit

File: console/src/main/java/org/eclipse/rdf4j/console/command/SetParameters.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (barthanssens) [reply console/src/main/java/org/eclipse/rdf4j/console/command/SetParameters.java]: Reworked this part, should be cleaner now

**Pull request merged**